code/game/objects/structures/tables_racks.dm

```diff
@@ -395,27 +395,30 @@
 		step(O, get_dir(O, src))
 	return
 
+/obj/structure/table/proc/tablepush(obj/item/I, mob/user)
+	if(get_dist(src, user) < 2)
+		var/obj/item/weapon/grab/G = I
+		if(G.affecting.buckled)
+			user << "<span class='warning'>[G.affecting] is buckled to [G.affecting.buckled]!</span>"
+			return 0
+		if(G.state < GRAB_AGGRESSIVE)
+			user << "<span class='warning'>You need a better grip to do that!</span>"
+			return 0
+		if(!G.confirm())
+			return 0
+		G.affecting.forceMove(get_turf(src))

```

Comment:
Random little fun fact; I accidentally made this forceMove(src).

People---people inside tables.